lib/menu.rb

```diff
@@ -0,0 +1,44 @@
+
+class Menu
```

Comment:
This class adheres well to SRP, it's succinctly written.

There are some over complexities - like each dish being available or not, but this doesn't affect the rest of the implementation.